server/block/button_type.go

```diff
@@ -0,0 +1,69 @@
+package block
+
+// ButtonType represents a type of button.
+type ButtonType struct {
+	button
+
+	// Wood is the type of wood of the button.
+	wood WoodType
+}
+
+type button uint8
+
+// WoodButton returns the wood button type.
+func WoodButton(w WoodType) ButtonType {
+	return ButtonType{0, w}
+}
+
+// StoneButton returns the stone button type.
+func StoneButton() ButtonType {
+	return ButtonType{button: 1}
+}
+
+// PolishedBlackstoneButton returns the polished blackstone button type.
+func PolishedBlackstoneButton() ButtonType {
+	return ButtonType{button: 2}
+}
+
+// Uint8 ...
+func (b ButtonType) Uint8() uint8 {
+	return b.wood.Uint8() | uint8(b.button)<<4
```

Comment:
This is not ideal. With the current order, as we get more wood variants, this will need to be updated, which might not be obvious to whoever is doing that.